grammar.js

```diff
@@ -535,6 +538,103 @@ module.exports = grammar({
 
     dotted_name: $ => sep1($.identifier, '.'),
 
+    // Match cases
+
+    case_pattern: $ => choice(
+      $.case_as_pattern,
+      $.case_or_pattern,
+    ),
+
+    case_as_pattern: $ => seq(
+      field('or_pattern', $.case_or_pattern), 'as', field('identifier', $.identifier)),
```

Comment:
So as I explained, these fields are unnecessary, the nodes are exposed and not inlined, and so it can only be matched in one spot in this rule.